### PR TITLE
Documentation: Treasury module

### PR DESCRIPTION
**Pull request by nczhu**
Draft writeup of Treasury module. 
Not ready for final review. 

Review courtesy @shawntabrizi

### CODE REVIEW

File: srml/treasury/src/lib.rs [hunk 1]
```diff
@@ -14,7 +14,48 @@
...
+//! The `treasury` module keeps account of currency in a `pot` and manages the subsequent
+//! deployment of these funds.
+//! 
+//! ## Overview
+//! 
+//! Funds for treasury are raised in two ways:
+//! 1. By minting new tokens, leading to inflation, and
+//! 2. By channelling some of the tokens set for burning from transaction fees and slashing.
+//! 
+//! Treasury funds can be used to pay for developers who provide software updates,
```
Comment (shawntabrizi) [inline srml/treasury/src/lib.rs]: Opinion: I think it makes sense for this module to have a **Scenarios** section similar to [staking](https://github.com/paritytech/substrate/blob/master/srml/staking/src/lib.rs#L49)

File: srml/treasury/src/lib.rs [hunk 2]
```diff
@@ -14,7 +14,48 @@
...
+//! 
+//! Please refer to the [`Call`](./enum.Call.html) enum and its associated variants for documentation on each function.
+//! 
+//! ### Public Functions
+//! 
+//! See the [module](./struct.Module.html) for details on publicly available functions.
+//! 
+//! ## Related Modules
+//! 
+//! The treasury module depends on the `system` and `srml_support` modules as well as
```
Comment (shawntabrizi) [inline srml/treasury/src/lib.rs]: This should be a bullet point list with links to the modules like:

https://github.com/paritytech/substrate/blob/master/srml/staking/src/lib.rs#L246

File: srml/treasury/src/lib.rs (hunk 1 above)
Comment (shawntabrizi) [reply srml/treasury/src/lib.rs]: In general, I think a module like this would fit well with the structure outlined in staking, although it is a bit simpler.

File: srml/treasury/src/lib.rs [hunk 3]
```diff
@@ -14,7 +14,48 @@
...
+//! # Treasury Module
+//! 
+//! The `treasury` module keeps account of currency in a `pot` and manages the subsequent
+//! deployment of these funds.
+//! 
+//! ## Overview
+//! 
+//! Funds for treasury are raised in two ways:
+//! 1. By minting new tokens, leading to inflation, and
+//! 2. By channelling some of the tokens set for burning from transaction fees and slashing.
```
Comment (joepetrowski) [inline srml/treasury/src/lib.rs]: Doesn't matter what they were meant for, and other modules may change what they are meant for.

```suggestion
//! 2. By channeling tokens from transaction fees and slashing.
```

File: srml/treasury/src/lib.rs [hunk 4]
```diff
@@ -14,7 +14,48 @@
...
+//! 
+//! Treasury funds can be used to pay for developers who provide software updates,
+//! any changes decided by referenda, and to generally keep the system running smoothly. 
+//! 
+//! Treasury can be used with other modules, such as to tax validator rewards in the `staking` module.
+//! 
+//! ### Implementations 
+//! 
+//! The treasury module provides an implementation for the following trait:
+//! - `OnDilution` - Mint extra funds upon dilution; maintain the ratio of `portion` diluted to `total_issuance`.
```
Comment (joepetrowski) [inline srml/treasury/src/lib.rs]: I don't think this should mention specific `variables`, especially ones that are generally handled outside this module. One can infer total issuance, but `portion` is a variable passed into this function and isn't defined anywhere here. Can you explain this trait in plain English?

File: srml/treasury/src/lib.rs [hunk 5]
```diff
@@ -14,7 +14,48 @@
...
+//! 
+//! - `propose_spend` - Propose a spending proposal and stake a proposal deposit.
+//! - `set_pot` - Set the spendable balance of funds.
+//! - `configure` - Configure the module's proposal requirements.
+//! - `reject_proposal` - Reject a proposal and slash the deposit.
+//! - `approve_proposal` - Accept the proposal and return the deposit.
+//! 
+//! Please refer to the [`Call`](./enum.Call.html) enum and its associated variants for documentation on each function.
+//! 
+//! ### Public Functions
```
Comment (joepetrowski) [inline srml/treasury/src/lib.rs]: This module doesn't have any public functions, so you can remove this section.

File: srml/treasury/src/lib.rs [hunk 6]
```diff
@@ -14,7 +14,48 @@
...
+//! 
+//! See the [module](./struct.Module.html) for details on publicly available functions.
+//! 
+//! ## Related Modules
```
Comment (joepetrowski) [inline srml/treasury/src/lib.rs]: You can remove this section. Everything depends on system and support, so we haven't been listing them. Same with Core.

**Comment by gavofyork** [comment]
Good additions but more detail needed on the transaction functions.
